claude-wrapper.sh logs [ERROR] for non-fatal auth timeout

## Summary

`check_auth_status()` in `claude-wrapper.sh` logs at `[ERROR]` level for **all** auth timeouts before the caller decides fatality. When the caller treats the timeout as non-fatal (e.g., in `--dangerously-skip-permissions` mode), the `[ERROR]` log line persists and triggers the shepherd's fallback pattern matching via `SYSTEMIC_FAILURE_PATTERNS`.

## Problem

The log level mismatch between the auth check function and its caller creates a false signal:

1. `check_auth_status()` always logs: `[ERROR] Authentication check timed out after 3 attempts`
2. Caller decides it's non-fatal, logs `[WARN]` and continues
3. Builder runs successfully, exits 0
4. Shepherd's `_is_auth_failure()` scans log, matches the `[ERROR]` from step 1
5. Shepherd misclassifies the run as an auth failure

## Fix Options

**Option A**: Have `check_auth_status()` return a status code and let the caller choose the log level.

**Option B**: Add a `[NON-FATAL]` prefix when in skip-permissions mode so the pattern doesn't match `SYSTEMIC_FAILURE_PATTERNS`.

## Impact

Contributing factor to Shepherd Run #2525 failure. Even with the exit-code guard fix, this log level inconsistency is a latent hazard.

## Relevant Code

- `claude-wrapper.sh` — `check_auth_status()` function
- `loom-tools/src/loom_tools/shepherd/phases/base.py:93-99` — `SYSTEMIC_FAILURE_PATTERNS`

--- a/defaults/scripts/claude-wrapper.sh
+++ b/defaults/scripts/claude-wrapper.sh
@@ -510,19 +510,19 @@ check_auth_status() {
                 sleep "${backoff}"
                 continue
             fi
-            log_error "Authentication check timed out after ${max_retries} attempts"
+            log_warn "Authentication check timed out after ${max_retries} attempts"
             [[ "${lock_acquired}" == "true" ]] && _auth_cache_unlock
             return 1
         fi
 
         if [[ "${auth_exit_code}" -ne 0 ]]; then
-            log_error "Authentication check command failed (exit ${auth_exit_code})"
-            log_error "Output: ${auth_output}"
+            log_warn "Authentication check command failed (exit ${auth_exit_code})"
+            log_warn "Output: ${auth_output}"
             if [[ -n "${CLAUDE_CONFIG_DIR:-}" ]]; then
-                log_error "CLAUDE_CONFIG_DIR=${CLAUDE_CONFIG_DIR}"
-                log_error "Run: CLAUDE_CONFIG_DIR=${CLAUDE_CONFIG_DIR} claude auth login"
+                log_warn "CLAUDE_CONFIG_DIR=${CLAUDE_CONFIG_DIR}"
+                log_warn "Run: CLAUDE_CONFIG_DIR=${CLAUDE_CONFIG_DIR} claude auth login"
             else
-                log_error "Run: claude auth login"
+                log_warn "Run: claude auth login"
             fi
             [[ "${lock_acquired}" == "true" ]] && _auth_cache_unlock
             return 1
@@ -536,12 +536,12 @@ check_auth_status() {
         logged_in=$(echo "${auth_output}" | python3 -c "import json,sys; print(json.load(sys.stdin).get('loggedIn', False))" 2>/dev/null || echo "")
 
         if [[ "${logged_in}" != "True" ]]; then
-            log_error "Authentication check failed: not logged in"
+            log_warn "Authentication check failed: not logged in"
             if [[ -n "${CLAUDE_CONFIG_DIR:-}" ]]; then
-                log_error "CLAUDE_CONFIG_DIR=${CLAUDE_CONFIG_DIR}"
-                log_error "Run: CLAUDE_CONFIG_DIR=${CLAUDE_CONFIG_DIR} claude auth login"
+                log_warn "CLAUDE_CONFIG_DIR=${CLAUDE_CONFIG_DIR}"
+                log_warn "Run: CLAUDE_CONFIG_DIR=${CLAUDE_CONFIG_DIR} claude auth login"
             else
-                log_error "Run: claude auth login"
+                log_warn "Run: claude auth login"
             fi
             [[ "${lock_acquired}" == "true" ]] && _auth_cache_unlock
             return 1
@@ -553,7 +553,7 @@ check_auth_status() {
     done
 
     # Should not reach here, but guard against it
-    log_error "Authentication check failed after ${max_retries} attempts"
+    log_warn "Authentication check failed after ${max_retries} attempts"
     [[ "${lock_acquired}" == "true" ]] && _auth_cache_unlock
     return 1
 }

--- a/loom-tools/src/loom_tools/shepherd/phases/base.py
+++ b/loom-tools/src/loom_tools/shepherd/phases/base.py
@@ -90,6 +90,11 @@ MCP_FAILURE_BACKOFF_SECONDS = [5, 15, 30]
 # that will NOT resolve with retries.  When detected after an instant-exit
 # or MCP failure, the shepherd should abort immediately instead of wasting
 # time on futile retry cycles.  See issue #2521.
+#
+# Note: check_auth_status() logs at [WARN] level (issue #2541), so only
+# the fatal caller path emits [ERROR].  The primary active pattern here is
+# "Authentication pre-flight check failed" (from run_preflight_checks).
+# Other auth patterns are retained as defense-in-depth.
 SYSTEMIC_FAILURE_PATTERNS = [
     re.compile(r"\[ERROR\]\s*Authentication check timed out", re.IGNORECASE),
     re.compile(r"\[ERROR\]\s*Authentication pre-flight check failed", re.IGNORECASE),

--- a/loom-tools/tests/shepherd/test_phases.py
+++ b/loom-tools/tests/shepherd/test_phases.py
@@ -14025,7 +14025,7 @@ class TestIsAuthFailure:
         log.write_text(
             "# Loom Agent Log\n"
             "[INFO] Running pre-flight checks...\n"
-            "[ERROR] Authentication check timed out after 3 attempts\n"
+            "[WARN] Authentication check timed out after 3 attempts\n"
             "[ERROR] Authentication pre-flight check failed\n"
             "# AUTH_PREFLIGHT_FAILED\n"
         )
@@ -14041,7 +14041,7 @@ class TestIsAuthFailure:
         assert _is_auth_failure(log) is True
 
     def test_pattern_fallback_auth_timed_out(self, tmp_path: Path) -> None:
-        """Auth failure detected via error pattern when sentinel is absent."""
+        """[ERROR] auth timeout still detected via pattern (defense-in-depth)."""
         log = tmp_path / "test.log"
         log.write_text(
             "# Loom Agent Log\n"
@@ -14075,6 +14075,38 @@ class TestIsAuthFailure:
             "[ERROR] Authentication check failed\n"
         )
         assert _is_auth_failure(log) is True
+
+    def test_warn_auth_timeout_not_classified(self, tmp_path: Path) -> None:
+        """[WARN] auth timeout (from check_auth_status) is NOT a systemic failure.
+
+        After issue #2541, check_auth_status() logs at [WARN] level.  Only the
+        fatal caller path emits [ERROR], so [WARN] messages should not trigger
+        _is_auth_failure().
+        """
+        log = tmp_path / "test.log"
+        log.write_text(
+            "# Loom Agent Log\n"
+            "[WARN] Authentication check timed out after 3 attempts\n"
+        )
+        assert _is_auth_failure(log) is False
+
+    def test_warn_auth_check_failed_not_classified(self, tmp_path: Path) -> None:
+        """[WARN] auth check failed (from check_auth_status) is NOT a systemic failure."""
+        log = tmp_path / "test.log"
+        log.write_text(
+            "# Loom Agent Log\n"
+            "[WARN] Authentication check failed: not logged in\n"
+        )
+        assert _is_auth_failure(log) is False
+
+    def test_warn_auth_command_failed_not_classified(self, tmp_path: Path) -> None:
+        """[WARN] auth command failed (from check_auth_status) is NOT a systemic failure."""
+        log = tmp_path / "test.log"
+        log.write_text(
+            "# Loom Agent Log\n"
+            "[WARN] Authentication check command failed (exit 1)\n"
+        )
+        assert _is_auth_failure(log) is False
 
     def test_no_false_positive_on_similar_text(self, tmp_path: Path) -> None:
         """Non-error messages containing similar words should not match."""

